From 6428d26f9e7516f611aa6871669247359ec469ae Mon Sep 17 00:00:00 2001
From: Andrea Canciani <ranma42@gmail.com>
Date: Fri, 4 May 2012 19:19:42 +0200
Subject: [PATCH 5/5] ntdll: Provide handle to trap deadlocks from debuggers

This adds a funciton with the purpose of making it possible to break
on it in debuggers, to make it easier to intercept deadlocks.
---
 dlls/ntdll/critsection.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/dlls/ntdll/critsection.c b/dlls/ntdll/critsection.c
index b865f67..96d24bc 100644
--- a/dlls/ntdll/critsection.c
+++ b/dlls/ntdll/critsection.c
@@ -409,6 +409,10 @@ NTSTATUS WINAPI RtlDeleteCriticalSection( RTL_CRITICAL_SECTION *crit )
     return STATUS_SUCCESS;
 }
 
+void RtlpWaitTimeout(void * crit)
+{
+    NTDLLCRIT_RTLP_WAIT_FOR_CRITICAL_SECTION_TIMEOUT(crit);
+}
 
 /***********************************************************************
  *           RtlpWaitForCriticalSection   (NTDLL.@)
@@ -441,7 +445,7 @@ NTSTATUS WINAPI RtlpWaitForCriticalSection( RTL_CRITICAL_SECTION *crit )
         if ( status == STATUS_TIMEOUT )
         {
             const char *name = NULL;
-	    NTDLLCRIT_RTLP_WAIT_FOR_CRITICAL_SECTION_TIMEOUT(crit);
+	    RtlpWaitTimeout(crit);
             if (crit->DebugInfo) name = (char *)crit->DebugInfo->Spare[0];
             if (!name) name = "?";
             ERR( "section %p %s wait timed out in thread %04x, blocked by %04x, retrying (60 sec)\n",
-- 
1.7.7.5 (Apple Git-26)

